src/main/java/slimeknights/tconstruct/plugin/jei/JEIPlugin.java

```diff
@@ -0,0 +1,65 @@
+package slimeknights.tconstruct.plugin.jei;
+
+import mezz.jei.api.IModPlugin;
+import mezz.jei.api.JeiPlugin;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.helpers.IJeiHelpers;
+import mezz.jei.api.registration.IRecipeCatalystRegistration;
+import mezz.jei.api.registration.IRecipeCategoryRegistration;
+import mezz.jei.api.registration.IRecipeRegistration;
+import net.minecraft.client.Minecraft;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.ResourceLocation;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.RecipeUtil;
+import slimeknights.tconstruct.library.recipe.casting.AbstractCastingRecipe;
+import slimeknights.tconstruct.plugin.jei.casting.CastingBasinCategory;
+import slimeknights.tconstruct.plugin.jei.casting.CastingTableCategory;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+
+@JeiPlugin
+public class JEIPlugin implements IModPlugin {
+
+  @Nullable
+  private CastingBasinCategory castingBasinCategory;
```

Comment:
Do we use these fields anywhere? Might as well make them local if not

Reply:
Don't think so, just modeled after JEI's furnace stuff.